app/models/dfe_sign_in_user.rb

```diff
@@ -14,13 +14,13 @@ def initialize(email:, dfe_sign_in_uid:, first_name:, last_name:, service:, id_t
 
   def self.begin_session!(session, omniauth_payload)
     session["dfe_sign_in_user"] = {
-      "email" => omniauth_payload["info"]["email"],
-      "dfe_sign_in_uid" => omniauth_payload["uid"],
-      "first_name" => omniauth_payload["info"]["first_name"],
-      "last_name" => omniauth_payload["info"]["last_name"],
+      "email" => omniauth_payload["info"]["email"].presence,
+      "dfe_sign_in_uid" => omniauth_payload["uid"].presence,
+      "first_name" => omniauth_payload["info"]["first_name"].presence,
+      "last_name" => omniauth_payload["info"]["last_name"].presence,
       "last_active_at" => Time.current,
-      "id_token" => omniauth_payload["credentials"]["id_token"],
-      "provider" => omniauth_payload["provider"],
+      "id_token" => omniauth_payload["credentials"]["id_token"].presence,
+      "provider" => omniauth_payload["provider"].presence,
```

Comment:
Having these attributes as nil will redirect the user to the sign in page with a message like You cannot access this service. In centry, there should be an error raised by dfe sign in

Reply:
I'm not sure I follow here. Why do we need to add the `.presence` call in again?

Reply:
It's for setting the attributes to nil if they are not returned by dfe sign in and this https://github.com/DFE-Digital/itt-mentor-services/blob/faf4e6f9a3a1944c0bfbf603bbaacc99715356bb/app/models/dfe_sign_in_user.rb#L61
Or this or this https://github.com/DFE-Digital/itt-mentor-services/blob/faf4e6f9a3a1944c0bfbf603bbaacc99715356bb/app/models/dfe_sign_in_user.rb#L68

Will return the user to the sign in page. Before, the code would just break here if the omniauth hash was not present

Reply:
Hmm, I've just removed the `.presence` calls and all the tests _still_ pass. Are we sure this is necessary?

Reply:
If it's necessary, can you add a test case that fails without the addition here please?

Reply:
Yeah, you're right, we don't need the `presence` here